### PR TITLE
CountVectorizerFeaturizer

### PR DESCRIPTION
**Pull request by angryjinyan**
CountVectorizerFeaturizer finished. Duplicate code removed.

### CODE REVIEW

File: src/Featurizers/CountVectorizerFeaturizer.h [hunk 1]
```diff
@@ -0,0 +1,451 @@
...
+void CountVectorizerTransformer::save(Archive &ar) const /*override*/ {
+    // Version
+    Traits<std::uint16_t>::serialize(ar, 1); // Major
+    Traits<std::uint16_t>::serialize(ar, 0); // Minor
+
+    // Data
+    Traits<decltype(_labels)>::serialize(ar, _labels);
+    Traits<decltype(_binary)>::serialize(ar, _binary);
+    Traits<decltype(_lower)>::serialize(ar, _lower);
+    Traits<std::uint8_t>::serialize(ar, static_cast<std::uint8_t>(_analyzer));
```
Comment (wangyems) [inline src/Featurizers/CountVectorizerFeaturizer.h]: Traits<std::underlying_type<AnalyzerMethod>::type>::serialize(ar, static_cast<std::underlying_type<AnalyzerMethod>::type>(_analyzer));

File: src/Featurizers/CountVectorizerFeaturizer.h [hunk 2]
```diff
@@ -0,0 +1,451 @@
...
+    typename BaseType::TransformerUniquePtr create_transformer_impl(void) override {
+
+        using DocumentStatisticsEstimator = Components::DocumentStatisticsEstimator<MaxNumTrainingItemsV>;
+
+        Components::DocumentStatisticsAnnotationData const &      data(DocumentStatisticsEstimator::get_annotation_data(BaseType::get_column_annotations(), _colIndex, Components::DocumentStatisticsEstimatorName));
+
+        typename Components::DocumentStatisticsAnnotationData::FrequencyAndIndexMap const &
+                                                            freqAndIndexes(data.TermFrequencyAndIndex);
+
+
```
Comment (michaelgsharp) [inline src/Featurizers/CountVectorizerFeaturizer.h]: You have 2 extra blank lines here. And another on line 209.

File: src/Featurizers/CountVectorizerFeaturizer.h [hunk 3]
```diff
@@ -0,0 +1,451 @@
...
+    return false;
+}
+
+template <size_t MaxNumTrainingItemsV>
+void Details::CountVectorizerEstimatorImpl<MaxNumTrainingItemsV>::complete_training_impl(void) /*override*/ {
+}
+
+}
+}
+}
```
Comment (michaelgsharp) [inline src/Featurizers/CountVectorizerFeaturizer.h]: Can you add the closing namespace comments here?

File: src/Featurizers/UnitTests/CountVectorizerFeaturizer_UnitTests.cpp [hunk 4]
```diff
@@ -0,0 +1,417 @@
...
+    std::vector<TransformedType::ValueEncoding> values;
+    values.emplace_back(TransformedType::ValueEncoding(3,0));
+    values.emplace_back(TransformedType::ValueEncoding(2,1));
+    values.emplace_back(TransformedType::ValueEncoding(1,2));
+    auto inferencingOutput = NS::TestHelpers::make_vector<TransformedType>(
+                                                                           TransformedType(3
+                                                                           , std::move(values))
+                                                                          );
+
+
```
Comment (michaelgsharp) [inline src/Featurizers/UnitTests/CountVectorizerFeaturizer_UnitTests.cpp]: You have an extra line before several of these `CHECK` calls.

**Pull request merged**